code/modules/mob/living/carbon/human/species/_species.dm

```diff
@@ -1033,16 +1033,15 @@ It'll return null if the organ doesn't correspond, so include null checks when u
 /datum/species/proc/spec_attacked_by(obj/item/I, mob/living/user, obj/item/organ/external/affecting, intent, mob/living/carbon/human/H)
 	return
 
-/proc/get_random_species(species_name = FALSE)	// Returns a random non black-listed or hazardous species, either as a string or datum
+/// Returns a list of names of non-blacklisted or hazardous species.
+/proc/get_safe_species()
 	var/static/list/random_species = list()
-	if(!random_species.len)
-		for(var/thing  in subtypesof(/datum/species))
+	if(!length(random_species))
+		for(var/thing in subtypesof(/datum/species))
 			var/datum/species/S = thing
 			if(!initial(S.dangerous_existence) && !initial(S.blacklisted))
 				random_species += initial(S.name)
```

Comment:
```suggestion
		for(var/datum/species/S in subtypesof(/datum/species))
			if(!initial(S.dangerous_existence) && !initial(S.blacklisted))
				random_species += initial(S.name)
```

Reply:
The contributor's guide says that if we know a container will always contain the same type/subtypes then the cast in the loop is faster: https://github.com/ParadiseSS13/Paradise/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops 

Does that apply here?

Reply:
Huh, I forgot that was in there. I'm pretty sure `as anything` should do the same thing as well?

Reply:
Pretty sure it does the same thing yeah, not doing the `istype()` on every item in the list. I'd prefer that as it is a bit cleaner to read